tests/e2e/scale/test_pods_are_not_oomkilled_while_running_ios.py

```diff
@@ -45,9 +46,29 @@ def base_setup(
         create maxsize pvc, pod and run IO
 
         """
-        osd_size = get_osd_size()
-        pvc_size_gb = osd_size * 1024
-        io_size_mb = f'{int((pvc_size_gb / 2) * 1000)}M'
+        # Setting the io_size_gb to 40% of the total PVC capacity
+        ceph_pod = Pod.get_ceph_tools_pod()
+        external = config.DEPLOYMENT['external_mode']
+        if external:
+            ocp_obj = ocp.OCP()
+            if interface == constants.CEPHBLOCKPOOL:
+                resource_name = constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_RBD
+            elif interface == constants.CEPHFILESYSTEM:
+                resource_name = constants.DEFAULT_EXTERNAL_MODE_STORAGECLASS_CEPHFS
+            cmd = f"get sc {resource_name} -o yaml"
+            pool_data = ocp_obj.exec_oc_cmd(cmd)
+            pool = pool_data['parameters']['pool']
+
+        else:
+            pool = constants.DEFAULT_BLOCKPOOL if interface == constants.CEPHBLOCKPOOL else constants.DATA_POOL
+
+        ceph_replica = ceph_pod.exec_ceph_cmd(ceph_cmd=f"ceph osd pool get {pool} size")
+        replica = ceph_replica['size']
+        ceph_status = ceph_pod.exec_ceph_cmd(ceph_cmd="ceph df")
+        ceph_capacity = int(ceph_status['stats']['total_bytes']) / replica / constants.GB
+        pvc_size_gb = int(ceph_capacity * 0.5)
+        io_size_gb = int(pvc_size_gb * 0.4)
```

Comment:
@ramkiperiy this is a shared cluster in case of external mode, should we reduce this a bit more or is it fine?

Reply:
Even if we get higher value, in next line there is a check to have io_size_gb not be more than 400G